src/components/UserInfoTable.tsx

```diff
@@ -0,0 +1,221 @@
+ */
+import { SearchOutlined } from '@ant-design/icons';
+import { Button, Input, Space, Table, Typography } from 'antd';
+import React, { useState } from 'react';
+import Highlighter from 'react-highlight-words';
+import { Link as RouterLink } from 'react-router-dom';
+import { PrivilegeLevel } from '../auth/ducks/types';
+const { Link } = Typography;
+
+export interface UserSummary {
+  firstName: string;
+  lastName: string;
+  email: string;
+  userId: number;
+  privilegeLevel: PrivilegeLevel;
+  phoneNumber: string;
+  profilePicture: string | null;
+  photoRelease: boolean;
+  detailsLink: string;
+}
+const TablePhotoConsent = (text: UserSummary, record: UserSummary) => (
+  <Space size="middle">{record.photoRelease ? 'Yes' : 'No'}</Space>
+);
+const TableActions = (text: UserSummary, record: UserSummary) => (
+  <Space size="middle">
+    <RouterLink to={record.detailsLink}>
+      <Link>View {record.firstName}'s Family Details</Link>
+    </RouterLink>
+  </Space>
+);
+
+interface UserInfoTableProps {
+  users: UserSummary[];
+}
+
+const UserInfoTable: React.FC<UserInfoTableProps> = ({ users }) => {
+  const [searchText, setSearchText] = useState<string>();
+  const [searchedColumn, setSearchedColumn] = useState<string>();
+
+  const getColumnSearchProps: (
+    dataIndex: string,
+  ) => {
+    filterDropdown: (props: {
+      setSelectedKeys: (value: string[]) => void;
+      selectedKeys: string[];
+      confirm: (props?: { closeDropdown: boolean }) => void;
+      clearFilters: () => void;
+    }) => JSX.Element;
+  } = (dataIndex: string) => ({
+    filterDropdown: ({
+      setSelectedKeys,
+      selectedKeys,
+      confirm,
+      clearFilters,
+    }) => (
+      <div style={{ padding: 8 }}>
+        <Input
+          placeholder={`Search ${dataIndex}`}
+          value={selectedKeys[0]}
+          onChange={(e) =>
+            setSelectedKeys(e.target.value ? [e.target.value] : [])
+          }
+          onPressEnter={() => handleSearch(selectedKeys, confirm, dataIndex)}
+          style={{ width: 188, marginBottom: 8, display: 'block' }}
+        />
+        <Space>
+          <Button
+            type="primary"
+            onClick={() => handleSearch(selectedKeys, confirm, dataIndex)}
+            icon={<SearchOutlined />}
+            size="small"
+            style={{ width: 90 }}
+          >
+            Search
+          </Button>
+          <Button
+            onClick={() => handleReset(clearFilters)}
+            size="small"
+            style={{ width: 90 }}
+          >
+            Reset
+          </Button>
+          <Button
+            type="link"
+            size="small"
+            onClick={() => {
+              confirm({ closeDropdown: false });
+              setSearchText(selectedKeys[0]);
+              setSearchedColumn(dataIndex);
+            }}
+          >
+            Filter
+          </Button>
+        </Space>
+      </div>
+    ),
+    filterIcon: (filtered: string) => (
```

Comment:
`(filtered: string): JSX.Element => (`

This suggestion applies to all of the other ones too

Reply:
I think this is ok because the type is implicit and I think its clear enough that it returns JSX, although im definitely willing to talk about it if you disagree